### PR TITLE
Pull in latest changes (including async allocatable components) from master

### PR DESCRIPTION
**Pull request by zbeekman**
This will pull in the latest changes from master, including allocatable components(WIP)

### CODE REVIEW

**Comment by zbeekman** [comment]
It looks like some conflicts need resolving.

File: prerequisites/build-functions/set_or_print_url.sh [hunk 1]
```diff
@@ -1,15 +1,22 @@
+# shellcheck shell=bash
 # If -p, -D, -P, or -V specifies a package, set package_url
 # If -U specifies a package, print the package_url and exit with normal status
 # shellcheck disable=SC2154
 set_or_print_url()
 {
   # Verify requirements
-  [ ! -z "${arg_U}" ] && [ ! -z "${arg_D:-${arg_p:-${arg_P:-${arg_V:-${arg_B}}}}}" ] &&
+  [[ -n "${arg_U}" && -n "${arg_D:-${arg_p:-${arg_P:-${arg_V:-${arg_B}}}}}" ]] &&
```
Comment (rouson) [inline prerequisites/build-functions/set_or_print_url.sh]: 👍

**Comment by rouson** [comment]
LGTM

**Comment by zbeekman** [comment]
Hi @rouson, I don't think you're commenting on the correct PR. This one needs some serious attention and is to get caffeinate opencoarrays up to date with master... it has fallen WAAAAAYYYYYYY behind.

I'm happy you approve of the changes I made to your PR though. They're showing up here because I went ahead and merged your PR into master since I figured my changes were small enough and comparatively trivial.

**Comment by zbeekman** [comment]
Closing PR due to stale-ness and discussion in [private SI gitter.im room](https://gitter.im/sourceryinstitute)

**Pull request closed without merging**